docs/open_source/scan/scan_vision/index.md

```diff
@@ -17,68 +15,339 @@ pip install giskard giskard-vision
+                'categorical_meta1': 'cat_value1',
+                'categorical_meta2': 'cat_value2'
+            },
+            categories=default_meta.categories+['categorical_meta1', 'categorical_meta2'],
+            issue_groups={
+                **default_meta.issue_groups,
+                'meta1': PerformanceIssueMeta,
+                'meta2': EthicalIssueMeta,
+                'categorical_meta1': PerformanceIssueMeta,
+                'categorical_meta2': EthicalIssueMeta,
+            }
+        )
+
+
+giskard_dataset = DataLoaderClassification()
+```
+::::::
+::::::{tab-item} Object detection
+
+To scan your model, start by **wrapping your dataset** with `DataIteratorBase`. Your class should override:
+
+- `idx_sampler` (property) that returns the array of indices to iterate over
+- `get_image` that loads the image corresponding to the right index as a `np.ndarray`
+- `get_label` that returns the box coordinates of the image with corresponding index
+- (Optional) `get_meta` that returns a `MetaData` object containing the metadata of the image with the corresponding index. This object should contain the metadata in the form of a `dict`, for instance `{'meta1': 'value1'}`, the list of categorical metadata, and the type of Issue related to the metadata in the form of a `dict`, for instance `{'meta1': PerformanceIssueMeta}`. Issue types can be found in `giskard_vision.core.issues`.
+
+> ### ⚠️ Warning
+>
+> It's highly recommended that you wrap your data **before preprocessing** so that you can easily interpret
+> the scan results.
+>
+> For now, the object detection scan only supports one box and label per image
+>
+
+```python
+from giskard_vision.core.dataloaders.base import DataIteratorBase
+from giskard_vision.core.dataloaders.meta import MetaData
+from giskard_vision.core.issues import EthicalIssueMeta, PerformanceIssueMeta
+
+
+class DataLoaderObjectDetection(DataIteratorBase):
+
+    @property
+    def idx_sampler(self) -> np.ndarray:
+        """Accesses the array of element indices to iterate over
+
+        Returns:
+            np.ndarray: The array of indices
+        """
+        return list(range(len(self.image_paths)))
+
+    @classmethod
+    def get_image(self, idx: int) -> np.ndarray:
+        """
+        Returns a single image from an index as numpy array
+
+        Args:
+            idx (int): Index of the image
+
+        Returns:
+            np.ndarray: The image as numpy array (h, w, c)
+        """
+        return self.images_paths[idx]
+
+    @classmethod
+    def get_label(self, idx: int) -> Optional[np.ndarray]:
+        """
+        Gets the box coordinates (for a single image) for a specific index.
+        Format:
+        np.array({"boxes": [x1_min, y1_min, x1_max, y1_max]})
+
+        Args:
+            idx (int): Index of the image.
+
+        Returns:
+            Optional[np.ndarray]: Box coordinates for the given index.
+        """
+        return np.array(..., dtype=object)
```

Comment:
it's rather something like this: https://github.com/Giskard-AI/giskard-vision/blob/46327872ebb54b5b28080f2ca388d9340b46e2b9/giskard_vision/object_detection/dataloaders/loaders.py#L235-L254